### PR TITLE
C++: Test changes for discover_walk

### PR DESCRIPTION
**Pull request by ian-semmle**

### CODE REVIEW

File: cpp/ql/test/library-tests/usings/Usings1.expected [hunk 1]
```diff
@@ -3,5 +3,4 @@
 | usings.cpp:9:1:9:17 | using namespace N |
 | usings.cpp:18:3:18:13 | using bf |
 | usings.cpp:21:5:21:14 | using gf |
-| usings.cpp:34:3:34:20 | using tbf |
```
Comment (geoffw0) [inline cpp/ql/test/library-tests/usings/Usings1.expected]: Any idea what's happened here?

File: cpp/ql/test/library-tests/templates/isfromtemplateinstantiation/isfromuninstantiatedtemplate.expected [hunk 2]
```diff
@@ -100,20 +96,37 @@ isFromUninstantiatedTemplate
 | isfromtemplateinstantiation.cpp:93:55:93:55 | return ... | isfromtemplateinstantiation.cpp:93:7:93:15 | myMethod1 |
 | isfromtemplateinstantiation.cpp:94:7:94:15 | declaration of myMethod2 | isfromtemplateinstantiation.cpp:77:26:77:45 | AnotherTemplateClass<T> |
 | isfromtemplateinstantiation.cpp:94:7:94:15 | declaration of myMethod2 | isfromtemplateinstantiation.cpp:97:25:97:60 | myMethod2 |
+| isfromtemplateinstantiation.cpp:94:7:94:15 | declaration of myMethod2 | isfromtemplateinstantiation.cpp:97:52:97:52 | myMethod2 |
```
Comment (geoffw0) [inline cpp/ql/test/library-tests/templates/isfromtemplateinstantiation/isfromuninstantiatedtemplate.expected]: I'd like @jbj to look at these changes at some point (no need to hold back merging though).

File: cpp/ql/test/library-tests/usings/Usings1.expected (hunk 1 above)
Comment (ian-semmle) [reply cpp/ql/test/library-tests/usings/Usings1.expected]: Fixed

File: cpp/ql/test/library-tests/templates/isfromtemplateinstantiation/isfromuninstantiatedtemplate.expected (hunk 2 above)
Comment (jbj) [reply cpp/ql/test/library-tests/templates/isfromtemplateinstantiation/isfromuninstantiatedtemplate.expected]: @ian-semmle and @nickrolfe, have you checked that the test changes in `isfromtemplateinstantiation/*.expected` are what they should be? These tests are very sensitive to exactly the things you changed in the discover branch, and they contain some of the edge cases that have been problematic in the past.

Comment (ian-semmle) [reply cpp/ql/test/library-tests/templates/isfromtemplateinstantiation/isfromuninstantiatedtemplate.expected]: The changes look OK to me. There's a new `| I | T |` row in isfromuninstantiatedtemplate, but that's probably the same issue as the existing rows.
